src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1438,11 +1438,25 @@ class Global(var currentSettings: Settings, reporter0: LegacyReporter)
     /** Caching member symbols that are def-s in Definitions because they might change from Run to Run. */
     val runDefinitions: definitions.RunDefinitions = new definitions.RunDefinitions
 
+    private def printArgs(sources: List[SourceFile]): Unit = {
+      val argsFile = (settings.recreateArgs ::: sources.map(_.file.absolute.toString())).mkString("", "\n", "\n")
+      settings.printArgs.value match {
+        case "-" =>
+          reporter.echo(argsFile)
+        case pathString =>
+          import java.nio.file._
+          val path = Paths.get(pathString)
+          Files.write(path, argsFile.getBytes(Charset.forName("UTF-8")))
```

Comment:
This decade there is [`StandardCharsets`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/charset/StandardCharsets.html).